### DISCUSSION TITLE
Page Break Issue in PDF Birt

### DISCUSSION THREAD

**Discussion by sharath492** [eclipse-birt/birt#1093] [Q&A]
Hello,

I am using BIRT Designer, I am having an issue of page break when generating the report as PDF.

After generating the PDF report, What happen is since in the 2nd page the header details
[rexel.pdf](https://github.com/eclipse/birt/files/9867809/rexel.pdf)
 are repeating same as 1st page but we need second masterPage details in 2 page pdf. 

I have tried many page break options like avoid before and after for tables/grid but nothing is working.

So Please give some suggestion to solve as soon as possible.

Please find the attachment of sample pdf for getting page break issue and rptdesign file and xml file.

Thanks & Regards,
Sharath
[Sharath.zip](https://github.com/eclipse/birt/files/9867828/Sharath.zip)
[rexel.pdf](https://github.com/eclipse/birt/files/9867833/rexel.pdf)

**Comment by speckyspooky** [comment]
Hello,

I have created an alternative szenario for you.
Your problem an the solution is a little bit tricky. The problem is that the master page is fixed at the table
and can only set 1times on a table which will work at the live-time of the table.

But a trick is to work with "pageNumber" and replace the header table from master page into the data table.
The next point is that the pageNumber will only work correctly in data detal rows but not on header rows (the page number on head rows is always 1).

In my example for you I added your head into the detail row an I work with 2 visibilitys to control your head.
1 visibility is located on header-grid directly and 1 visibility is your 2nd header row.

grid-visibility controll that the head will be created only 1times on a page
and the row-visibility controls that the row will be display only on page one.

For the correct work, I created an page-variable wich will be rest on new pages (after page break).
And this variable will be set on row.onCreate() which will be used as Hidden-flag-value for your head.

Attached ypu will find my report and the pdf output based on your example.

Only thing is, I wasn't able to create more like 1 page with your original data table.
Therefore I used an own tabled but with your data-xml-source.

I hope it will help you to find an solution for your final report.

[rexelaccpaypdf-nlpurchaseorderfromb2bexml.zip](https://github.com/eclipse/birt/files/9873767/rexelaccpaypdf-nlpurchaseorderfromb2bexml.zip)
[rexelaccpaypdf-nlpurchaseorderfromb2bexml-TEST.pdf](https://github.com/eclipse/birt/files/9873769/rexelaccpaypdf-nlpurchaseorderfromb2bexml-TEST.pdf)

**Reply by sharath492** [reply]
Hello [speckyspooky](https://github.com/speckyspooky)

Thank you for your suggestion :).

But the attached rpt design was not opened in my eclipse application. My eclipse version is 3.2.23 but your attached file is 3.2.24.

can you please suggest how to open in my 3.2.23 version?.

**Reply by speckyspooky** [reply]
Hello sharath92,

the report from yesterday is created with BIRT 4.10.
Here you get the version with BIRT 4.8, I hope it will help you.

[rexelaccpaypdf-nlpurchaseorderfromb2bexml_birt_48.zip](https://github.com/eclipse/birt/files/9882082/rexelaccpaypdf-nlpurchaseorderfromb2bexml_birt_48.zip)

**Reply by sharath492** [reply]
Hello [speckyspooky](https://github.com/speckyspooky),

The report with Birt 4.8 is opened in my application. 

Thank you for your help.

**Comment by hvbtup** [comment]
Does this solution require separate RunTask and RenderTask or does it also work with RunAndRenderTask?

**Comment by hvbtup** [comment]
An alternative approach in similar situations, somehow similar to @speckyspooky 's idea, is like this:

It uses only one masterpage and doesn't need page variables.

Use a table header row and make sure "repeat header" is *not* checked in the table's page break properties.

Place some content A into the table header row.

Add a group to the table - this group might also be a "dummy", e.g. use a constant like `1` as the group expression (instead of `row["..."]`).

A table group header row will be added automatically.

Place some content B into the table group header row.

Depending on your needs, consider adding a visibility expression like `row.__rownum <=0` to that group header row.
Depending on your needs, set or don't set "repeat header" in the page break properties of the table group.

The content A will be shown ...
* on the first page only, if "repeat header" is not checked for the table.
* on all pages (if "repeat header" is checked for the table.

The content B will be shown ...
* on on all but the first pages, if "repeat header" is checked for the group and due to the visibility expression.
* on all pages, if "repeat header" is checked for the group and you don't specify a visibility expression.
* on the first page only, if "repeat header" is not checked for the group and you don't specify a visibility expression.

This way, you can control the rendering of A and B independently.

**Reply by sharath492** [reply]
Hi [hvbtup](https://github.com/hvbtup),

Thank you for your valuable response.
